### PR TITLE
Updated project configuration checker window

### PR DESCRIPTION
**Pull request by Troy-Ferrell**
## Overview
It has been a difficult balance for MRTK to control setting up Unity projects to help with configuration, performance, and expected functionality without annoying or frustrating users. We want this configuration to mostly be automated but also inform developers and give them control to select themselves. 

This change revamps the configuration check dialog generally seen when adding MRTK to a new project. It creates it into a popup utility window that allows for a better user experience. Developers can now see and control more clearly what will be automated to their project. They can also ignore the popup for later (which will hide it for the rest of the current Unity session) or Ignore it entirely (which will set a user preference to never show the window).

image

Popup primarily shows on re-compiles, opened Unity projects, or changing build targets.

Developers can also access this popup via the MRTK > Utilities tab directly.

Note: This is not placed in the Tools package because it is more essential to the MRTK experience. Some users do not download the Tools package. 

Created MRProjectConfigurator and MRProjectConfiguratorWindow
Moved MixedRealityEditorSettings file to new folder
Simplified MREditorSettings fucntionality to just track responsive to editor states (i.e play mode, etc)
Moved directory utilities out of MREditorSettings and into EditorProjectUtilities
Introduced capability checking in list

## Changes
- Fixes: #6175 
Related: #4194

### CODE REVIEW

**Comment by Troy-Ferrell** [comment]
@davidkline-ms, @keveleigh , or @Cameron-Micka , can you review this once any of you get a chance? It's about to hit 10 days online

File: Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs [hunk 1]
```diff
@@ -0,0 +1,249 @@
...
+        /// <summary>
+        /// Configures current Unity project to contain the default Spatial Awareness layer set in the Layers settings
+        /// </summary>
+        public static void SetSpatialAwarenessLayer()
+        {
+            if (!HasSpatialAwarenessLayer())
+            {
+                if (EditorLayerExtensions.SetupLayer(SpatialAwarenessDefaultLayer, "Spatial Awareness"))
+                {
+                    Debug.LogWarning(string.Format($"Can't modify project layers. It's possible the format of the layers and tags data has changed in this version of Unity. Set layer {SpatialAwarenessDefaultLayer} to \"Spatial Awareness\" manually via Project Settings > Tags and Layers window."));
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs]: This message is misleading if the layer is already set to Spatial Awareness.  Can we get a wording tweak or have it succeed if the value is already set?

Another option would be to differentiate "failure to set", "already set" and "set to something we didnt expect"

**Comment by davidkline-ms** [comment]
Also seeing the first run not display the logo image

image

This is a fresh 2018.4.11f1 created project, importing a test version of the foundation package generated from @Troy-Ferrell's fork.

File: Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs [hunk 2]
```diff
@@ -0,0 +1,249 @@
...
+        }
+
+        /// <summary>
+        /// Configures current Unity project to contain the default Spatial Awareness layer set in the Layers settings
+        /// </summary>
+        public static void SetSpatialAwarenessLayer()
+        {
+            if (!HasSpatialAwarenessLayer())
+            {
+                if (EditorLayerExtensions.SetupLayer(SpatialAwarenessDefaultLayer, "Spatial Awareness"))
```
Comment (davidkline-ms) [inline Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs]:
```suggestion
                if (!EditorLayerExtensions.SetupLayer(SpatialAwarenessDefaultLayer, "Spatial Awareness"))
```

Comment (davidkline-ms) [reply Assets/MixedRealityToolkit/Utilities/Editor/Setup/MixedRealityProjectConfigurator.cs]: just noticed that this check appears to be backwards :)

**Comment by Troy-Ferrell** [comment]
> fork

Interesting, @keveleigh also saw something somewhat similar in Unity 2019.3.

I wonder if it's a timing issue in the way Unity loads texture...or at least the way we are doing it

**Comment by keveleigh** [comment]
Weird on the reported logo issue! I didn't see it, but I also didn't import from scratch. Interesting...

**Comment by davidkline-ms** [comment]
- confirmed that the spatial layer no longer incorrectly appears.
- First run (on package import) still shows no image
- After the process completed, a package loaded and the dialog re-appeared (this was confusing as the settings are already applied)

**Comment by davidkline-ms** [comment]
once merged, i will file separate issues based on what i encountered.

**Pull request merged**